src/tree/containerizedFunctionApp/FunctionsTreeItem.ts

```diff
@@ -0,0 +1,104 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { type FunctionEnvelope, type Site } from "@azure/arm-appservice";
+import { createWebSiteClient } from "@microsoft/vscode-azext-azureappservice";
+import { uiUtils } from "@microsoft/vscode-azext-azureutils";
+import { nonNullProp, nonNullValueAndProp, type AzExtTreeItem, type IActionContext } from "@microsoft/vscode-azext-utils";
+import * as retry from 'p-retry';
+import { ParsedFunctionJson } from "../../funcConfig/function";
+import { localize } from "../../localize";
+import { FunctionsTreeItemBase } from "../FunctionsTreeItemBase";
+import { getFunctionNameFromId } from "../remoteProject/RemoteFunctionTreeItem";
+import { type ContainerTreeItem } from "./ContainerTreeItem";
+import { FunctionItem } from "./FunctionItem";
+import { FunctionTreeItem } from "./FunctionTreeItem";
+
+export class FunctionsTreeItem extends FunctionsTreeItemBase {
+    public isReadOnly: boolean = true;
+    private _nextLink: string | undefined;
+
+    constructor(public readonly parent: ContainerTreeItem, public readonly site: Site) {
+        super(parent);
+    }
+
+    public static async createFunctionsTreeItem(context: IActionContext, parent: ContainerTreeItem): Promise<FunctionsTreeItem> {
+        const ti: FunctionsTreeItem = new FunctionsTreeItem(parent, parent.site);
+        // initialize
+        await ti.refreshImpl(context);
+        return ti;
+    }
+
+    public hasMoreChildrenImpl(): boolean {
+        return !!this._nextLink;
+    }
+
+    public async refreshImpl(context: IActionContext): Promise<void> {
+        this.isReadOnly = await this.parent.isReadOnly(context);
```

Comment:
You probably don't need to do this as the containerized functions should always be read-only, right?